### PR TITLE
Cruft link

### PR DESCRIPTION
**Pull request by CyclingNinja**

### CODE REVIEW

File: .cruft.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      "use_compiled_extensions": "n",
+      "enable_dynamic_dev_versions": "n",
+      "include_example_code": "n",
+      "_sphinx_theme": "alabaster",
+      "_parent_project": "",
+      "_install_requires": "",
+      "_copy_without_render": [
+        "docs/_templates",
+        "docs/_static",
+        ".github/workflows/sub_package_update.yml"
```
Comment (Cadair) [inline .cruft.json:24]:
```suggestion
        "docs/_static"
```

File: docs/conf.py [hunk 2]
```diff
@@ -1,30 +1,31 @@
...
 """
 # flake8: NOQA: E402
 
-# -- stdlib imports ------------------------------------------------------------
-from parfive import __version__
-import datetime
-from packaging.version import Version
-
-# -- Project information -------------------------------------------------------
+# -- Project information -----------------------------------------------------
```
Comment (Cadair) [inline docs/conf.py:13]:
```suggestion
# -- stdlib imports ------------------------------------------------------------
import datetime
from packaging.version import Version

# -- Project information -----------------------------------------------------
```

File: docs/conf.py [hunk 3]
```diff
@@ -1,30 +1,31 @@
...
+#
+# Configuration file for the Sphinx documentation builder.
+#
+# This file does only contain a selection of the most common options. For a
+# full list see the documentation:
+# http://www.sphinx-doc.org/en/master/config
 
 isort:skip_file
 """
 # flake8: NOQA: E402
```
Comment (Cadair) [inline docs/conf.py:12]:
```suggestion
```

File: pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,98 @@
...
-line-length = 101
+]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "parfive"
+description = "A HTTP and FTP parallel file downloader."
+readme = "README.rst"
+requires-python = ">=3.7"
+license = { file = "licenses/LICENSE.rst", content-type = "text/plain" }
```
Comment (Cadair) [inline pyproject.toml:15]:
```suggestion
license = { file = "LICENSE", content-type = "text/plain" }
```

File: pyproject.toml [hunk 5]
```diff
@@ -1,6 +1,98 @@
...
+test = [
+    "pytest",
+    "pytest-doctestplus",
+]
+docs = [
+    "sphinx",
+    "sphinx-automodapi",
+    "tomli; python_version <\"3.11\"",
+]
+
```
Comment (Cadair) [inline pyproject.toml:36]: why is the template so empty line happy here, we should address that upstream

File: pyproject.toml [hunk 6]
```diff
@@ -1,6 +1,98 @@
...
+
+
+
+[project.urls]
+repository = "https://parfive.readthedocs.io/"
```
Comment (Cadair) [inline pyproject.toml:41]: This is wrong, looks like a template bug

File: pyproject.toml [hunk 7]
```diff
@@ -1,6 +1,98 @@
...
+  # Don't complain if tests don't hit assertions
+  "raise AssertionError",
+  "raise NotImplementedError",
+  # Don't complain about script hooks
+  "def main(.*):",
+  # Ignore branches that don't pertain to this version of Python
+  "pragma: py{ignore_python_version}",
+  # Don't complain about IPython completion helper
+  "def _ipython_key_completions_",
+]
```
Comment (Cadair) [inline pyproject.toml:98]:
```suggestion
]

[tool.black]
line-length = 101
```

File: setup.py [hunk 8]
```diff
@@ -1,7 +1,4 @@
 #!/usr/bin/env python
 from setuptools import setup
 
-
-setup(
-
-)
+setup()
```
Comment (Cadair) [inline setup.py:4]: @CyclingNinja This change should also be fixed in the template. rendering with a blank line like that is gross.

Comment (CyclingNinja) [reply setup.py]: Yeah agreed, I'm wondering how it even happened

File: pyproject.toml (hunk 6 above)
Comment (CyclingNinja) [reply pyproject.toml:38]: thats the url to the docs not the repo

Comment (CyclingNinja) [reply pyproject.toml:38]: I'm not sure what's happened here, all the template stuff looks to be in good order, did I just mess up the input?

Comment (Cadair) [reply pyproject.toml:38]: It should be `Homepage =`

File: pyproject.toml [hunk 9]
```diff
@@ -1,6 +1,131 @@
...
+include_trailing_comma = true
+known_first_party = "parfive"
+length_sort = false
+length_sort_sections = "stdlib"
+line_length = 110
+multi_line_output = 3
+no_lines_before = "LOCALFOLDER"
+sections = "STDLIB, THIRDPARTY, FIRSTPARTY, LOCALFOLDER"
+
+[tool.mypy]
```
Comment (nabobalis) [inline pyproject.toml:103]: @Cadair TYPING IN MY PARFIVE?

Comment (Cadair) [reply pyproject.toml:103]: I just form my opinions out of nowhere :stuck_out_tongue:

**Pull request closed without merging**